flocker/control/_model.py

```diff
@@ -609,6 +609,10 @@ def expire(self, now):
         return updated
 
 
+class PersistentState(PClass):
```

Comment:
I've documented the general purpose. The specific purpose for this series of PRs is to have place to record a mapping from `dataset_id`s to `blockdevice_id`s. This is going to supersede the metadata we are currently using to discover this, since the eventual consistency of our backends means we might create duplicates and then randomly pick different volumes depending on the precise results from the backend.

This won't actually prevent duplicates from being created, but it will prevent them from being used.